actions/notification_actions.jsx

```diff
@@ -17,121 +17,132 @@ import {stripMarkdown} from 'utils/markdown';
+            }
+        }
+
+        let title = Utils.localizeMessage('channel_loader.posted', 'Posted');
+        if (!channel) {
             title = msgProps.channel_display_name;
+            channel = {
+                name: msgProps.channel_name,
+                type: msgProps.channel_type,
+            };
+        } else if (channel.type === Constants.DM_CHANNEL) {
+            title = Utils.localizeMessage('notification.dm', 'Direct Message');
+        } else {
+            title = channel.display_name;
+        }
+
+        if (title === '') {
+            if (msgProps.channel_type === Constants.DM_CHANNEL) {
+                title = Utils.localizeMessage('notification.dm', 'Direct Message');
+            } else {
+                title = msgProps.channel_display_name;
+            }
         }
-    }
-
-    let notifyText = post.message;
-
-    const msgPropsPost = JSON.parse(msgProps.post);
-    const attachments = msgPropsPost && msgPropsPost.props && msgPropsPost.props.attachments ? msgPropsPost.props.attachments : [];
-    let image = false;
-    attachments.forEach((attachment) => {
-        if (notifyText.length === 0) {
-            notifyText = attachment.fallback ||
-                         attachment.pretext ||
-                         attachment.text;
+
+        let notifyText = post.message;
+
+        const msgPropsPost = JSON.parse(msgProps.post);
+        const attachments = msgPropsPost && msgPropsPost.props && msgPropsPost.props.attachments ? msgPropsPost.props.attachments : [];
+        let image = false;
+        attachments.forEach((attachment) => {
+            if (notifyText.length === 0) {
+                notifyText = attachment.fallback ||
+                    attachment.pretext ||
+                    attachment.text;
+            }
+            image |= attachment.image_url.length > 0;
+        });
+
+        let strippedMarkdownNotifyText = stripMarkdown(notifyText);
+        if (strippedMarkdownNotifyText.length > NOTIFY_TEXT_MAX_LENGTH) {
+            strippedMarkdownNotifyText = strippedMarkdownNotifyText.substring(0, NOTIFY_TEXT_MAX_LENGTH - 1) + '...';
         }
-        image |= attachment.image_url.length > 0;
-    });
-
-    let strippedMarkdownNotifyText = stripMarkdown(notifyText);
-    if (strippedMarkdownNotifyText.length > NOTIFY_TEXT_MAX_LENGTH) {
-        strippedMarkdownNotifyText = strippedMarkdownNotifyText.substring(0, NOTIFY_TEXT_MAX_LENGTH - 1) + '...';
-    }
-
-    let body = '';
-    if (strippedMarkdownNotifyText.length === 0) {
-        if (msgProps.image) {
-            body = username + Utils.localizeMessage('channel_loader.uploadedImage', ' uploaded an image');
-        } else if (msgProps.otherFile) {
-            body = username + Utils.localizeMessage('channel_loader.uploadedFile', ' uploaded a file');
-        } else if (image) {
-            body = username + Utils.localizeMessage('channel_loader.postedImage', ' posted an image');
+
+        let body = `@${username}: `;
```

Comment:
Are we intentionally removing the `@user wrote:`? That seems like it may have been a design oversight

Reply:
yes this is intentional, it matches the push notification